### PR TITLE
chore(api): add ExamEnvironmentAuthorizationToken -> user relation

### PR DESCRIPTION
**Pull request by ShaunSHamilton**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/how-to-open-a-pull-request/).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Originally, there was a constraint to keep all exam tokens. However, I cannot remember the exact reason for this. So, this also removes it, and we will have to just see if the constraint resurfaces.

### CODE REVIEW

File: api/prisma/schema.prisma [hunk 1]
```diff
@@ -375,15 +376,13 @@ model UserToken {
...
 }
 
-/// TODO: Token has to outlive the exam attempt
-/// Validation has to be taken as the attempt is requested
-/// to ensure it lives long enough.
 model ExamEnvironmentAuthorizationToken {
   id          String   @id @map("_id")
   createdDate DateTime @db.Date
-  userId      String   @db.ObjectId
+  userId      String   @unique @db.ObjectId
```
Comment (ojeytonwilliams) [inline api/prisma/schema.prisma:382]: In order to impose this constraint we should `prisma db push` or create the constraint by hand. That applies to the prod db as well as the tests.

**Review by ojeytonwilliams — CHANGES_REQUESTED** [review]
Why not not include this in the user object? I know we tend to dump a lot of stuff in there, but it seems unnecessary.

The only reason I can think of is that it might save us a request. So, if we can demonstrate that it's causing perf issues, we can do it. Otherwise I don't think it makes sense.

File: api/prisma/schema.prisma (hunk 1 above)
Comment (ShaunSHamilton) [reply api/prisma/schema.prisma:382]: > Why not not include this in the user object?

Include what in the `user` object? This PR includes the token in the `user` object.

> In order to impose this constraint we should prisma db push or create the constraint by hand.

What are you saying? This is a new collection (does not exist yet). So, the constraint cannot cause issues?

Comment (ojeytonwilliams) [reply api/prisma/schema.prisma:382]: > Include what in the user object? This PR includes the token in the user object.

Yeah, that. There's already an endpoint to get the token, so why put it in the user object?

> What are you saying? This is a new collection (does not exist yet). So, the constraint cannot cause issues?

That adding the constraint to the schema does nothing on its own - it has to be applied to the dbs (e.g. via `prisma db push`) to take effet.

Comment (ShaunSHamilton) [reply api/prisma/schema.prisma:382]: > There's already an endpoint to get the token, so why put it in the user object?

Oh. Sorry. My brain ignored the second _"not"_ in your sentence.

Reason: There is no endpoint to get the token. There is only an endpoint to generate it.

Would you just create an endpoint to GET/generate it?

Comment (ShaunSHamilton) [reply api/prisma/schema.prisma:382]: >  it has to be applied to the dbs

Ah. I thought you were suggesting we needed a migration to use the constraint.

Comment (ShaunSHamilton) [reply api/prisma/schema.prisma:382]: Very true. Thank you.

File: api/src/routes/protected/user.test.ts [hunk 2]
```diff
@@ -1130,6 +1136,72 @@ Thanks and regards,
...
+        const response = await superPost('/user/exam-environment/token');
+
+        const { examEnvironmentAuthorizationToken } = response.body.data;
+
+        const decodedToken = jwt.decode(examEnvironmentAuthorizationToken);
+
+        expect(decodedToken).not.toStrictEqual({
+          examEnvironmentAuthorizationToken: id,
+          iat: expect.any(Number)
+        });
```
Comment (ojeytonwilliams) [inline api/src/routes/protected/user.test.ts:1192]:
```suggestion
        expect(decodedToken).not.toHaveProperty(
          'examEnvironmentAuthorizationToken',
          id
        );
```
Nit: this seems better aligned with what we're interested in (that the token has updated).

**Review by ojeytonwilliams — APPROVED** [review]
Only one nit, so I'm approving :+1:

**Review by Sembauke — APPROVED** [review]

**Pull request merged**